### PR TITLE
[external-dns] Support ServiceSource

### PR DESCRIPTION
**Pull request by zoetrope**
Signed-off-by: zoetrope <a.ikezoe@gmail.com>

### CODE REVIEW

**Review by masa213f — APPROVED** [review]
LGTM

File: test/metallb_test.go [hunk 1]
```diff
@@ -119,5 +119,11 @@ func testMetalLB() {
 		Eventually(func() error {
 			return exec.Command("ip", "netns", "exec", "external", "curl", targetIP, "-m", "5").Run()
 		}).Should(Succeed())
+
+		By("access service from external with fqdn")
+		Eventually(func() error {
+			return exec.Command("ip", "netns", "exec", "external", "curl", "http://testhttpd.default.gcp0.dev-ne.co", "-m", "5").Run()
```
Comment (binoue) [inline test/metallb_test.go:125]: Can we resolve dns in the external ns?

Comment (zoetrope) [reply test/metallb_test.go]: I tried, but I couldn't resolve it.
Why can't I do that?

**Review by masa213f — APPROVED** [review]
LGTM

**Pull request merged**